daemon/libnetwork/portallocator/osallocator_linux.go

```diff
@@ -158,8 +226,21 @@ func bindTCPOrUDP(addr netip.AddrPort, typ int, proto types.Protocol) (_ *os.Fil
 	return boundSocket, nil
 }
 
-// bindSCTP is based on sctp.ListenSCTP. The socket is created and bound, but
-// does not start listening.
+// listenSCTP is based on sctp.ListenSCTP.
+func listenSCTP(addr netip.AddrPort) (_ *os.File, retErr error) {
+	boundSocket, err := bindSCTP(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	somaxconn := -1 // silently capped to "/proc/sys/net/core/somaxconn"
+	if err := syscall.Listen(int(boundSocket.Fd()), somaxconn); err != nil {
+		return nil, fmt.Errorf("failed to listen on sctp socket: %w", err)
+	}
+
+	return boundSocket, nil
+}
+
 func bindSCTP(addr netip.AddrPort) (_ *os.File, retErr error) {
```

Comment:
Did we lose the GoDoc for `bindSCTP` ?

Reply:
Honestly I don't see the point of these comments. It's just going to be redundant with other comments, and it's quite obvious that `listenSCTP` creates a listening SCTP socket, and `bindSCTP` creates a non-listening SCTP socket. But I can add one back if you want.